### PR TITLE
Change comment style

### PR DESCRIPTION
**Pull request by Diadlo**
1. Small refactoring
2. Docs comments moved in cpp file (where it's possible)
3. Added doxygen.conf
- [X] We need to generate html docs by doxygen?
- [X] What you think about adding rule "All new function should has comment" to contributing?

### CODE REVIEW

File: src/video/videomode.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#include "videomode.h"
```
Comment (zetok) [inline src/video/videomode.cpp:20]: Needs copyright header

**Comment by zetok** [comment]
> What you think about adding rule "All new function should has comment" to contributing?

Yes.

Would "All new code has to have comments" be better?

> We need to generate html docs by doxygen?

Yep. Additionally, travis could be set to push docs to the website when building master branch.

**Comment by Diadlo** [comment]
> Additionally, travis could be set to push docs to the website when building master branch.

It's perfect idea. Can you help with it after merge?

**Comment by zetok** [comment]
> It's perfect idea. Can you help with it after merge?

Yep.

> https://github.com/tux3/qTox/pull/3536/commits/0a1101d3541b9e03bcdbd84078ed43881889025d

No, remove that.

Also, add `doc/html/*` to `.gitignore`.

**Comment by zetok** [comment]
6009929bb75aa0af9f721f2ad99bc2246df23e7e – `.h` still seems to contain the implementation?
https://github.com/Diadlo/qTox/blob/6009929bb75aa0af9f721f2ad99bc2246df23e7e/src/video/videomode.h

Also, copyright on .cpp should be at least `2015`, i.e. starting from the the year when the code in question was written (date that is in `.h` file), + if code was modified, later copyright dates can be added, so that it could be represented as date range, e.g. `2015-2016`.

Date when code was just copypasted doesn't really matter.

**Comment by zetok** [comment]
PR doesn't build, needs to be fixed

**Comment by zetok** [comment]
PR needs to be rebased, since there are now some conflicts.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -197,6 +197,57 @@ QObject notToMentionThatWeUseCamelCase;
 
 E.g. https://github.com/tux3/qTox/blob/master/src/misc/flowlayout.cpp
 
+## Documenataion
+
+If you added new function, add doxygen comment before implementation.
```
Comment (sudden6) [inline CONTRIBUTING.md]: `If you added a new function, also add a doxygen comment before the implementation.
If you changed an old function, make sure the doxygen comment is still correct. If it doesn't exist add it.`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -197,6 +197,57 @@ QObject notToMentionThatWeUseCamelCase;
...
+If you changed old function, be sure, that doxygen comment still correct. If it
+doesn't exist by any reason, add it.
+
+Don't put docs in .h file, if same .cpp file exist.
```
Comment (sudden6) [inline CONTRIBUTING.md]: `Don't put docs in .h files, if there is a corresponding .cpp file.`

File: src/audio/audio.cpp [hunk 4]
```diff
@@ -88,7 +86,30 @@ class Audio::Private
 };
 
 /**
-Returns the singleton instance.
+@class Audio
+
+@fn void Audio::frameAvailable(const int16_t *pcm, size_t sample_count, uint8_t channels, uint32_t sampling_rate);
+
+When there are input subscribers, we regularly emit captured audio frames with this signal
+Always connect with a blocking queued connection or a lambda, or the behavior is undefined
```
Comment (sudden6) [inline src/audio/audio.cpp]: I suggest `lambda, else the behaviour`

File: src/core/coreav.cpp [hunk 5]
```diff
@@ -31,7 +31,52 @@
 #include <QCoreApplication>
 #include <QtConcurrent/QtConcurrentRun>
 
+/**
+@fn void CoreAV::avInvite(uint32_t friendId, bool video)
+@brief Sent when a friend calls us.
+@param friendId Id of friend in call list.
+@param video Does't this invite in video chat?
```
Comment (sudden6) [inline src/core/coreav.cpp]: I don't understand this line, maybe reword?

File: src/core/coreav.cpp [hunk 6]
```diff
@@ -31,7 +31,52 @@
...
+
+@fn void CoreAV::avStart(uint32_t friendId, bool video)
+@brief Sent when a call we initiated has started.
+@param friendId Id of friend in call list.
+@param video Does't this video chat started?
```
Comment (sudden6) [inline src/core/coreav.cpp]: same

File: src/core/coreencryption.cpp [hunk 7]
```diff
@@ -51,6 +51,7 @@ std::unique_ptr<TOX_PASS_KEY> Core::createPasskey(const QString& password, uint8
     return encryptionKey;
 }
 
+///< Uses the default profile's key
```
Comment (sudden6) [inline src/core/coreencryption.cpp]: default or currently active?

File: src/core/coreencryption.cpp [hunk 8]
```diff
@@ -68,6 +69,7 @@ QByteArray Core::encryptData(const QByteArray& data, const TOX_PASS_KEY& encrypt
     return QByteArray(reinterpret_cast<char*>(encrypted), data.size() + TOX_PASS_ENCRYPTION_EXTRA_LENGTH);
 }
 
+///< Uses the default profile's key
```
Comment (sudden6) [inline src/core/coreencryption.cpp]: same

File: src/core/corefile.cpp [hunk 9]
```diff
@@ -30,14 +30,27 @@
...
+
 QMutex CoreFile::fileSendMutex;
 QHash<uint64_t, ToxFile> CoreFile::fileMap;
 using namespace std;
 
+/**
+@brief Get corefile iteration interval.
+
+tox_iterate calls to get good file transfer performances
+@return The maximum amount of time in ms that Core should wait between two.
```
Comment (sudden6) [inline src/core/corefile.cpp]: two what?

File: src/core/toxcall.cpp [hunk 10]
```diff
@@ -7,6 +7,23 @@
...
 
+/**
+@var uint32_t ToxCall::callId
+@brief Could be a friendNum or groupNum, must uniquely identify the call. Do not modify!
+
+@var bool ToxCall::inactive
+@brief True while we're not participating. (stopped group call, ringing but hasn't started yet, ...)
+
+@var bool ToxCall::videoEnabled
+@brief True if our user asked for a video call, sending and recving
```
Comment (sudden6) [inline src/core/toxcall.cpp]: typo: recieving

File: src/core/toxid.cpp [hunk 11]
```diff
@@ -52,33 +84,60 @@ ToxId::ToxId(const QString &id)
...
+*/
 bool ToxId::operator==(const ToxId& other) const
 {
     return publicKey == other.publicKey;
 }
 
+/**
+@brief Compares, that only public key not equals.
+@param other Tox ID to compare.
+@return True if both Tox ID has different public key, false otherwise.
```
Comment (sudden6) [inline src/core/toxid.cpp]: has different public key -> have different public keys

File: src/core/toxid.cpp [hunk 12]
```diff
@@ -52,33 +84,60 @@ ToxId::ToxId(const QString &id)
...
+*/
 bool ToxId::operator!=(const ToxId &other) const
 {
     return publicKey != other.publicKey;
 }
 
+/**
+@brief Check, that current user ID.
```
Comment (sudden6) [inline src/core/toxid.cpp]: `Check, that the current user ID  is the active user ID` ?

File: src/core/toxid.cpp [hunk 13]
```diff
@@ -52,33 +84,60 @@ ToxId::ToxId(const QString &id)
...
+*/
 void ToxId::clear()
 {
     publicKey.clear();
     noSpam.clear();
     checkSum.clear();
 }
 
+/**
+@brief Check, that id is valid Tox ID.
```
Comment (sudden6) [inline src/core/toxid.cpp]: is a valid

File: src/ipc.cpp [hunk 14]
```diff
@@ -84,13 +97,23 @@ IPC::~IPC()
...
     return instance;
 }
 
-time_t IPC::postEvent(const QString &name, const QByteArray& data/*=QByteArray()*/, uint32_t dest/*=0*/)
+/**
+@brief Post IPC event.
+@param name Name to set in IPC event.
+@param data Data to set in IPC event (default QByteArray()).
+@param dest Settings::getCurrentProfileId() or 0 (main instance, default).
+@return Time of new last event.
```
Comment (sudden6) [inline src/ipc.cpp]: I don't understand this comment

File: src/net/autoupdate.cpp [hunk 15]
```diff
@@ -511,6 +625,14 @@ void AutoUpdater::installLocalUpdate()
     exit(0);
 }
 
+/**
+@brief Checks update an show dialog asking to download it.
+@note Runs asynchronously in its own thread, and will return immediatly
+
+Will call isUpdateAvailable, and as such may processEvents.
+Connects to the qTox update server, if an updat is found
```
Comment (sudden6) [inline src/net/autoupdate.cpp]: typo: update

File: src/net/toxdns.cpp [hunk 16]
```diff
@@ -104,13 +124,20 @@ QByteArray ToxDNS::fetchLastTextRecord(const QString& record, bool silent)
...
-    return result;
+    return textRecordValues.first();
 }
 
+/**
+@brief Send query to DNS to find Tox Id.
+@note Will *NOT* fallback on queryTox1 anymore.
+@param server Server to sending query.
+@param record Should look like user@domain.tld.
+@param silent If true, there will no output when error.
```
Comment (sudden6) [inline src/net/toxdns.cpp]: will be no output on error

File: src/net/toxdns.cpp [hunk 17]
```diff
@@ -190,44 +217,40 @@ QString ToxDNS::queryTox3(const tox3_server& server, const QString &record, bool
     return toxIdStr;
 }
 
+/**
+@brief Tries to map a text string to a ToxId struct, will query Tox DNS records if necessary.
+@param address Address to search tox id.
```
Comment (sudden6) [inline src/net/toxdns.cpp]: Adress to search for Tox ID.

File: src/net/toxdns.cpp [hunk 18]
```diff
@@ -190,44 +217,40 @@ QString ToxDNS::queryTox3(const tox3_server& server, const QString &record, bool
...
+@param silent If true, there will no output when error.
```
Comment (sudden6) [inline src/net/toxdns.cpp]: will be no output on error

File: src/net/toxdns.cpp [hunk 19]
```diff
@@ -190,44 +217,40 @@ QString ToxDNS::queryTox3(const tox3_server& server, const QString &record, bool
...
 {
-    ToxId toxId;
-
     if (address.isEmpty())
+        return ToxId();
+
+    if (ToxId::isToxId(address))
+        return ToxId(address);
+
+    // If we're querying one of our pinned server, do a toxdns3 request directly
```
Comment (sudden6) [inline src/net/toxdns.cpp]: servers

File: src/net/toxme.cpp [hunk 20]
```diff
@@ -204,6 +217,17 @@ Toxme::ExecCode Toxme::extractError(QString json)
     return ExecCode(r);
 }
 
+/**
+@brief Creates a new toxme address associated with a Tox ID.
+@param[out] code Tox error code @see getErrorMessage.
+@param[in] server Server to creating toxme account.
```
Comment (sudden6) [inline src/net/toxme.cpp]: Create toxme account on this server.

File: src/net/toxme.cpp [hunk 21]
```diff
@@ -204,6 +217,17 @@ Toxme::ExecCode Toxme::extractError(QString json)
...
+@param[in] id ToxId of current user.
+@param[in] address Addres to creating toxme account.
```
Comment (sudden6) [inline src/net/toxme.cpp]: Create toxme account with this adress.

File: src/net/toxme.cpp [hunk 22]
```diff
@@ -204,6 +217,17 @@ Toxme::ExecCode Toxme::extractError(QString json)
...
+@param[in] keepPrivate If true, the address will not be published on toxme site.
+@param[in] bio A short optional description of yourself if you want to publish your address.
+@return password, if it passed without error,
```
Comment (sudden6) [inline src/net/toxme.cpp]: password on success, else sets code parameter and returns an empty QString.

File: src/net/toxme.cpp [hunk 23]
```diff
@@ -271,6 +295,12 @@ QString Toxme::getPass(QString json, ExecCode &code) {
     return json;
 }
 
+/**
+@brief Deletes the address associated with your current Tox ID.
+@param server Server to deleting address.
```
Comment (sudden6) [inline src/net/toxme.cpp]: Server to delete the address from.

File: src/net/toxme.cpp [hunk 24]
```diff
@@ -271,6 +295,12 @@ QString Toxme::getPass(QString json, ExecCode &code) {
...
+@param id ToxId to deleting.
```
Comment (sudden6) [inline src/net/toxme.cpp]: ToxId to delete.

File: src/net/toxme.cpp [hunk 25]
```diff
@@ -271,6 +295,12 @@ QString Toxme::getPass(QString json, ExecCode &code) {
...
+@return Code returned from server.
```
Comment (sudden6) [inline src/net/toxme.cpp]: Status code returned from server

File: src/net/toxuri.cpp [hunk 26]
```diff
@@ -44,6 +44,12 @@ bool toxURIEventHandler(const QByteArray& eventData)
     return true;
 }
 
+/**
+@brief Shows a dialog asking whether or not to add this tox address as a friend.
+@note Will wait until the core is ready first.
+@param toxURI Tox URI to trying add.
```
Comment (sudden6) [inline src/net/toxuri.cpp]: to try to add

File: src/net/toxuri.cpp [hunk 27]
```diff
@@ -44,6 +44,12 @@ bool toxURIEventHandler(const QByteArray& eventData)
...
+@return True, if tox URI are correct, false otherwise.
```
Comment (sudden6) [inline src/net/toxuri.cpp]: are -> is

File: src/nexus.cpp [hunk 28]
```diff
@@ -131,6 +144,9 @@ void Nexus::start()
         showLogin();
 }
 
+/**
+@brief Hides the man GUI, delete the profile, and shows the login screen
```
Comment (sudden6) [inline src/nexus.cpp]: man -> main, deletes the profile and shows

File: src/nexus.cpp [hunk 29]
```diff
@@ -224,18 +247,30 @@ Core* Nexus::getCore()
...
+@brief Get current user profile.
+@return nullptr if not started, profile otherwise.
+*/
 Profile* Nexus::getProfile()
 {
     return getInstance().profile;
 }
 
+/**
+@brief Delete the current profile, if any, and replaces it.
```
Comment (sudden6) [inline src/nexus.cpp]: delete -> unload ?

File: src/nexus.cpp [hunk 30]
```diff
@@ -250,6 +285,11 @@ QString Nexus::getSupportedImageFilter()
   return tr("Images (%1)", "filetype filter").arg(res.left(res.size()-1));
 }
 
+/**
+@brief Dangerous way to find out if a path is writable.
+@param filepath Path to file which should be deleted.
+@return True, if file witable, false otherwise.
```
Comment (sudden6) [inline src/nexus.cpp]: typo: writeable

File: src/persistence/db/rawdatabase.cpp [hunk 31]
```diff
@@ -14,6 +14,60 @@
...
+@brief If not a nullptr, the result of the transaction will be set
+
+@var std::atomic_bool* RawDatabase::Transaction::done = nullptr;
+@brief If not a nullptr, will be set to true when the transaction has been executed
+*/
+
+/**
+@brief Tries to open a database.
+@param path Path to database.
+@param password If is empty, the database will be opened unencrypted.
```
Comment (sudden6) [inline src/persistence/db/rawdatabase.cpp]: `If empty` ?

File: src/persistence/db/rawdatabase.cpp [hunk 32]
```diff
@@ -75,6 +135,9 @@ bool RawDatabase::open(const QString& path, const QString &hexKey)
     return true;
 }
 
+/**
+@brief Closes the database and free its associated resources.
```
Comment (sudden6) [inline src/persistence/db/rawdatabase.cpp]: Closes -> Close

File: src/persistence/db/rawdatabase.cpp [hunk 33]
```diff
@@ -89,22 +152,41 @@ void RawDatabase::close()
         qWarning() << "Error closing database:"<<sqlite3_errmsg(sqlite);
 }
 
+/**
+@brief Checks, that database is opened
```
Comment (sudden6) [inline src/persistence/db/rawdatabase.cpp]: that the database is open

File: src/persistence/db/rawdatabase.cpp [hunk 34]
```diff
@@ -89,22 +152,41 @@ void RawDatabase::close()
...
+@return Whether the transaction was successful.
+*/
 bool RawDatabase::execNow(const RawDatabase::Query &statement)
 {
     return execNow(QVector<Query>{statement});
 }
 
+/**
+@brief Executes a SQL transaction synchronously.
+@param statements List of statement to execute
```
Comment (sudden6) [inline src/persistence/db/rawdatabase.cpp]: statement -> statements

File: src/persistence/history.cpp [hunk 35]
```diff
@@ -8,12 +8,32 @@
 
 using namespace std;
 
+/**
+@class History
+@brief Interacts with the profile database to save the chat history.
+
+@var QHash<QString, int64_t> History::peers
+@brief Maps friend public keys to unique IDs by index.
+Cached mappings to speed up message saving.
```
Comment (sudden6) [inline src/persistence/history.cpp]: Caches

File: src/persistence/offlinemsgengine.cpp [hunk 36]
```diff
@@ -26,6 +26,15 @@
 #include <QMutexLocker>
 #include <QTimer>
 
+/**
+@var static const int OfflineMsgEngine::offlineTimeout
+@brief timeout after which faux offline messages get to be re-sent.
+Originally was 2s, but since that was causing lots of duplicated
+messages on receiving end, make qTox be more lazy about re-sending
+should be 20.
```
Comment (sudden6) [inline src/persistence/offlinemsgengine.cpp]: 20s

File: src/persistence/profile.cpp [hunk 37]
```diff
@@ -65,6 +80,14 @@ Profile::Profile(QString name, const QString &password, bool isNewProfile)
     QObject::connect(coreThread, &QThread::started, core, &Core::start);
 }
 
+/**
+@brief Locks and loads an existing profile and create the associate Core* instance.
```
Comment (sudden6) [inline src/persistence/profile.cpp]: creates

File: src/persistence/profile.cpp [hunk 38]
```diff
@@ -340,6 +396,12 @@ void Profile::saveToxSave(QByteArray data)
     }
 }
 
+/**
+@brief Gets the path of the avatar file cached by this profile and corresponding to this owner ID.
+@param ownerId Path to avatar of friend with this ID will returned.
+@param forceUnencrypted If forceUnencrypted, we return the path to the plaintext file even if we're an encrypted profile.
```
Comment (sudden6) [inline src/persistence/profile.cpp]: If true, return the path to the plaintext file even if this is an encrypted profile.

File: src/persistence/profile.cpp [hunk 39]
```diff
@@ -433,21 +525,36 @@ QByteArray Profile::getAvatarHash(const QString &ownerId)
...
+@brief Removes our own avatar.
+*/
 void Profile::removeAvatar()
 {
     removeAvatar(core->getSelfId().publicKey);
 }
 
+/**
+@brief Checks that the history is enabled in the settings, and loaded successfully for this profile.
+@return True if alright, false otherwise.
```
Comment (sudden6) [inline src/persistence/profile.cpp]: True if enabled, false otherwise.

File: src/persistence/profile.cpp [hunk 40]
```diff
@@ -433,21 +525,36 @@ QByteArray Profile::getAvatarHash(const QString &ownerId)
...
+@return May return a nullptr if the history failed to load.
+*/
 History *Profile::getHistory()
 {
     return history.get();
 }
 
+/**
+@brief Removes a cached avatar.
+@param ownerId Friend IF wich avatar was deleted.
```
Comment (sudden6) [inline src/persistence/profile.cpp]: Friend ID whose avater to delete.

File: src/persistence/profile.cpp [hunk 41]
```diff
@@ -461,11 +568,21 @@ bool Profile::exists(QString name)
     return QFile::exists(path+".tox");
 }
 
+/**
+@brief Checks, that profile has password.
```
Comment (sudden6) [inline src/persistence/profile.cpp]: Checks, if profile has a password.

File: src/persistence/profile.cpp [hunk 42]
```diff
@@ -461,11 +568,21 @@ bool Profile::exists(QString name)
...
+@return True if we have a password set (doesn't check the actual file on disk).
+*/
 bool Profile::isEncrypted() const
 {
     return !password.isEmpty();
 }
 
+/**
+@brief Chacks, that profile is encrypted.
```
Comment (sudden6) [inline src/persistence/profile.cpp]: Checks if profile is encrypted.

File: src/persistence/profile.cpp [hunk 43]
```diff
@@ -461,11 +568,21 @@ bool Profile::exists(QString name)
...
+@note Checks the actual file on disk.
+@param name Profile name.
+@return False on error, true otherwise.
```
Comment (sudden6) [inline src/persistence/profile.cpp]: `True if profile is encrypted, false otherwise` ?

File: src/persistence/profile.cpp [hunk 44]
```diff
@@ -546,6 +669,11 @@ QVector<QString> Profile::remove()
     return ret;
 }
 
+/**
+@brief Tries to rename the profile.
+@param newName New name to profile.
```
Comment (sudden6) [inline src/persistence/profile.cpp]: to -> for the

File: src/persistence/profile.cpp [hunk 45]
```diff
@@ -594,6 +729,10 @@ void Profile::restartCore()
     QMetaObject::invokeMethod(core, "reset");
 }
 
+/**
+@brief Changes the encryption password and re-saves everything with it
+@param newPassword Password to envryption.
```
Comment (sudden6) [inline src/persistence/profile.cpp]: Password for encryption

File: src/persistence/profilelocker.cpp [hunk 46]
```diff
@@ -62,16 +83,26 @@ bool ProfileLocker::lock(QString profile)
...
     delete lockfile.release();
     lockfile = nullptr;
     curLockName.clear();
 }
 
+/**
+@brief Check that we actually own the lock.
+In case the file was deleted on disk, restore it.
+If we can't get a lock, exit qTox immediately.
+If we never had a lock in the firt place, exit immediately.
```
Comment (sudden6) [inline src/persistence/profilelocker.cpp]: typo: first

File: src/persistence/settings.cpp [hunk 47]
```diff
@@ -600,6 +624,10 @@ uint32_t Settings::makeProfileId(const QString& profile)
     return dwords[0] ^ dwords[1] ^ dwords[2] ^ dwords[3];
 }
 
+/**
+@brief Get path to directory, where stored settings files.
```
Comment (sudden6) [inline src/persistence/settings.cpp]: where the settings files are stored.

File: src/persistence/settings.cpp [hunk 48]
```diff
@@ -600,6 +624,10 @@ uint32_t Settings::makeProfileId(const QString& profile)
...
+@return Path to settings directory ends with a directory separator.
```
Comment (sudden6) [inline src/persistence/settings.cpp]: directory, ends

File: src/persistence/settings.cpp [hunk 49]
```diff
@@ -619,6 +647,10 @@ QString Settings::getSettingsDirPath()
 #endif
 }
 
+/**
+@brief Get path to directory, where stored application data.
```
Comment (sudden6) [inline src/persistence/settings.cpp]: same as above

File: src/persistence/settings.cpp [hunk 50]
```diff
@@ -640,6 +672,10 @@ QString Settings::getAppDataDirPath()
 #endif
 }
 
+/**
+@brief Get path to directory, where stored application cache.
+@return Path to application cache ends with a directory separator.
```
Comment (sudden6) [inline src/persistence/settings.cpp]: same as above

File: src/persistence/settings.cpp [hunk 51]
```diff
@@ -1846,6 +1882,11 @@ void Settings::setAutoLogin(bool state)
     autoLogin = state;
 }
 
+/**
+@brief Write a default personal .ini settings file for a profile.
+@param basename Filename without extention to save settings.
```
Comment (sudden6) [inline src/persistence/settings.cpp]: typo: extension

File: src/persistence/settingsserializer.cpp [hunk 52]
```diff
@@ -28,8 +28,37 @@
...
+SettingsSerializer can detect regular .ini files and serialized ones,
+it will read both regular and serialized .ini, but only save in serialized format.
+The file is encrypted with the current profile's password, if any.
+The file is only written to disk if save() is called, the destructor does not save to disk
+All member functions are reentrant, but not thread safe.
+
+@enum SettingsSerializer::RecordTag
+@var Value
+Followed by a QString key then a QVariant value
+@var GroupStart,
```
Comment (sudden6) [inline src/persistence/settingsserializer.cpp]: `,` doesn't belong here I think

File: src/platform/statusnotifier/statusnotifier.c [hunk 53]
```diff
@@ -501,15 +501,15 @@ status_notifier_class_init (StatusNotifierClass *klass)
...
+    @sn: The #StatusNotifier
+    @x: screen coordinates X
+    @y: screen coordinates Y
      *
      * Emitted when the ContextMenu method was called on the item. Item should
      * then show a context menu, this is typically a consequence of user input,
      * such as mouse right click over the graphical representation of the item.
      *
-     * @x and @y are to be considered an hint to the item about where to show
+    @x and @y are to be considered an hint to the item about where to show
```
Comment (sudden6) [inline src/platform/statusnotifier/statusnotifier.c]: a hint

File: src/platform/statusnotifier/statusnotifier.c [hunk 54]
```diff
@@ -527,15 +527,15 @@ status_notifier_class_init (StatusNotifierClass *klass)
...
+    @sn: The #StatusNotifier
+    @x: screen coordinates X
+    @y: screen coordinates Y
      *
      * Emitted when the Activate method was called on the item. Activation of
      * the item was requested, this is typically a consequence of user input,
      * such as mouse left click over the graphical representation of the item.
      *
-     * @x and @y are to be considered an hint to the item about where to show
+    @x and @y are to be considered an hint to the item about where to show
```
Comment (sudden6) [inline src/platform/statusnotifier/statusnotifier.c]: a hint

File: src/platform/statusnotifier/statusnotifier.c [hunk 55]
```diff
@@ -553,17 +553,17 @@ status_notifier_class_init (StatusNotifierClass *klass)
...
+    @y: screen coordinates Y
      *
      * Emitted when the SecondaryActivate method was called on the item.
      * Secondary and less important form of activation (compared to
      * #StatusNotifier::activate) of the item was requested. This is typically a
      * consequence of user input, such as mouse middle click over the graphical
      * representation of the item.
      *
-     * @x and @y are to be considered an hint to the item about where to show
+    @x and @y are to be considered an hint to the item about where to show
```
Comment (sudden6) [inline src/platform/statusnotifier/statusnotifier.c]: a hint

File: src/video/cameradevice.cpp [hunk 56]
```diff
@@ -205,11 +239,20 @@ CameraDevice* CameraDevice::open(QString devName, VideoMode mode)
...
 void CameraDevice::open()
 {
     ++refcount;
 }
 
+/**
+@brief Closes the device. Never fails.
+@note If returns true, "this" becomes invalid.
+@return True, if device finally deleted (closed last reference),
+false otherwise (if exist other reference).
```
Comment (sudden6) [inline src/video/cameradevice.cpp]: (if other references exist)

File: src/video/cameradevice.cpp [hunk 57]
```diff
@@ -376,6 +445,11 @@ QVector<VideoMode> CameraDevice::getScreenModes()
     return result;
 }
 
+/**
+@brief Get the list of video modes for a device.
+@param devName Device name to getting modes.
```
Comment (sudden6) [inline src/video/cameradevice.cpp]: to get nodes from

File: src/video/cameradevice.cpp [hunk 58]
```diff
@@ -376,6 +445,11 @@ QVector<VideoMode> CameraDevice::getScreenModes()
...
+@return Vector of avaliable moves to the device.
```
Comment (sudden6) [inline src/video/cameradevice.cpp]: available modes for the device

File: src/video/cameradevice.cpp [hunk 59]
```diff
@@ -401,6 +475,11 @@ QVector<VideoMode> CameraDevice::getVideoModes(QString devName)
     return {};
 }
 
+/**
+@brief Get the name of the pixel format of a video mode.
+@param pixel_format Pixel format to getting name.
```
Comment (sudden6) [inline src/video/cameradevice.cpp]: to get the name from

File: src/video/cameradevice.cpp [hunk 60]
```diff
@@ -401,6 +475,11 @@ QVector<VideoMode> CameraDevice::getVideoModes(QString devName)
...
+@return Name of the pixe format.
```
Comment (sudden6) [inline src/video/cameradevice.cpp]: pixel

File: src/video/cameradevice.cpp [hunk 61]
```diff
@@ -410,6 +489,13 @@ QString CameraDevice::getPixelFormatString(uint32_t pixel_format)
 #endif
 }
 
+/**
+@brief Compare two pixel format.
```
Comment (sudden6) [inline src/video/cameradevice.cpp]: formats

File: src/video/corevideosource.cpp [hunk 62]
```diff
@@ -25,12 +25,34 @@ extern "C" {
...
+*/
 CoreVideoSource::CoreVideoSource()
     : subscribers{0}, deleteOnClose{false},
     stopped{false}
 {
 }
 
+/**
+@brief Makes a copy of the vpx_image_t and emits it as a new VideoFrame.
+@param vpxframe Frame to copying.
```
Comment (sudden6) [inline src/video/corevideosource.cpp]: to copy

File: src/video/corevideosource.cpp [hunk 63]
```diff
@@ -108,12 +130,22 @@ void CoreVideoSource::unsubscribe()
     biglock.unlock();
 }
 
+/**
+@brief Setup deleting on close
```
Comment (sudden6) [inline src/video/corevideosource.cpp]: deleting -> delete

File: src/video/videoframe.cpp [hunk 64]
```diff
@@ -78,6 +93,11 @@ VideoFrame::~VideoFrame()
     releaseFrameLockless();
 }
 
+/**
+@brief Converts the VideoFrame to a QImage that shares our internal video buffer.
+@param size Size of result image.
```
Comment (sudden6) [inline src/video/videoframe.cpp]: resulting

File: src/video/videomode.cpp [hunk 65]
```diff
@@ -0,0 +1,75 @@
...
+@brief Describes a video mode supported by a device.
+
+@var unsigned short VideoMode::width, VideoMode::height
+@brief Displayed video resolution (NOT frame resolution).
+
+@var unsigned short VideoMode::x, VideoMode::y
+@brief Coordinates of upper-left corner.
+
+@var float VideoMode::FPS
+@brief Max frames per second supported by the device at this resolution
```
Comment (sudden6) [inline src/video/videomode.cpp]: I don't think it's the max fps here

File: src/video/videosurface.cpp [hunk 66]
```diff
@@ -63,6 +68,13 @@ bool VideoSurface::isExpanding() const
     return expanding;
 }
 
+/**
+@brief Update source
+@note nullptr is a valid option
+@param src source to settings
```
Comment (sudden6) [inline src/video/videosurface.cpp]: source to set

File: src/widget/about/aboutuser.cpp [hunk 67]
```diff
@@ -89,9 +89,8 @@ void AboutUser::onSelectDirClicked()
 }
 
 /**
- * @brief AboutUser::onAcceptedClicked When users clicks the bottom OK button,
- *          save all settings
- */
+@brief Called when users clicks the bottom OK button, save all settings
```
Comment (sudden6) [inline src/widget/about/aboutuser.cpp]: users -> user

File: src/widget/form/tabcompleter.cpp [hunk 68]
```diff
@@ -20,15 +20,21 @@
     along with qTox.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-/* This file was taken from the Quassel IRC client source (src/uisupport), and
-   was greatly simplified for use in qTox. */
+/**
+@file tabcompleter.h
+@file tabcompleter.cpp
+This files was taken from the Quassel IRC client source (src/uisupport), and
```
Comment (sudden6) [inline src/widget/form/tabcompleter.cpp]: `files -> file` ?

File: src/widget/notificationscrollarea.cpp [hunk 69]
```diff
@@ -67,16 +67,16 @@ void NotificationScrollArea::trackWidget(GenericChatroomWidget* widget)
 }
 
 /**
- * @brief Delete notification bar to visible elements on scroll area
- */
+@brief Delete notification bar to visible elements on scroll area
```
Comment (sudden6) [inline src/widget/notificationscrollarea.cpp]: bar from visible

File: src/widget/tool/screenshotgrabber.cpp [hunk 70]
```diff
@@ -192,10 +192,9 @@ void ScreenshotGrabber::chooseHelperTooltipText(QRect rect)
 }
 
 /**
- * @internal
- *
- * Align the tooltip centred at top of screen with the mouse cursor.
- */
+@internal
+@brief Align the tooltip centred at top of screen with the mouse cursor.
```
Comment (sudden6) [inline src/widget/tool/screenshotgrabber.cpp]: centered

**Comment by sudden6** [comment]
done with review

File: src/core/coreav.cpp (hunk 5 above)
Comment (Diadlo) [inline src/core/coreav.cpp]: If you get invited in chat with video, `video` must be true.
If you get invited in audio-only chat, `video` must be false.
What I should write?

File: src/ipc.cpp (hunk 14 above)
Comment (Diadlo) [inline src/ipc.cpp]: `Time of event finish` better?

File: src/widget/tool/screenshotgrabber.cpp (hunk 70 above)
Comment (zetok) [inline src/widget/tool/screenshotgrabber.cpp]: actually, both forms are correct, depending on the english version (gb/us)

File: src/platform/statusnotifier/statusnotifier.c (hunk 53 above)
Comment (Diadlo) [inline src/platform/statusnotifier/statusnotifier.c]: Oh.. My mistake. I not sure, that should modify `statusnotifier.c`

File: src/widget/tool/screenshotgrabber.cpp (hunk 70 above)
Comment (sudden6) [inline src/widget/tool/screenshotgrabber.cpp]: ok, I'm more used to the british style :). Then this doesn't need to be changed.

File: src/widget/form/tabcompleter.cpp (hunk 68 above)
Comment (Diadlo) [inline src/widget/form/tabcompleter.cpp]: Not.

> @file tabcompleter.h
> @file tabcompleter.cpp

Comment (sudden6) [inline src/widget/form/tabcompleter.cpp]: Then it's:

`These files were taken`

File: src/core/coreav.cpp (hunk 5 above)
Comment (sudden6) [inline src/core/coreav.cpp]: `False if chat is audio only, true audio and video.`

File: src/core/coreav.cpp (hunk 6 above)
Comment (sudden6) [inline src/core/coreav.cpp]: `False if chat is audio only, true audio and video.`

File: src/ipc.cpp (hunk 14 above)
Comment (sudden6) [inline src/ipc.cpp]: Do you mean `Time the event finished` ?

**Comment by Diadlo** [comment]
@sudden6 Thank you very much, for this review!

**Comment by sudden6** [comment]
@Diadlo thank you very much
